buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -929,6 +930,16 @@ private void deallocate() {
             }
         }
 
+        private static boolean shouldReleaseSuboptimalChunkSuze(int givenSize, int preferredSize) {
+            int givenChunks = givenSize / MIN_CHUNK_SIZE;
+            int preferredChunks = preferredSize / MIN_CHUNK_SIZE;
+            int deviation = Math.abs(givenChunks - preferredChunks);
+
+            // Retire chunks with a 0.5% probability per unit of MIN_CHUNK_SIZE deviation from preference.
+            return deviation != 0 &&
+                    PlatformDependent.threadLocalRandom().nextDouble() * 200.0 > deviation;
```

Comment:
I have to check how it will behave with virtual threads - literally, no idea

Reply:
Pretty sure `ThreadLocalRandom` (the real one) has special access to the platform thread.